semantic-java/src/Language/Java/Tags.hs

```diff
@@ -76,6 +80,24 @@ instance ToTags Java.MethodInvocation where
       Tags.yield (Tag name Call loc (Tags.firstLine src range) Nothing)
       gtags t
 
+instance ToTags Java.InterfaceDeclaration where
+  tags t@Java.InterfaceDeclaration
+    { ann = loc@Loc { byteRange = Range { start } }
+    , name = Java.Identifier { text = name }
+    , body = Java.InterfaceBody { ann = Loc Range { start = end } _ }
+    } = do
+      src <- ask @Source
+      Tags.yield (Tag name Interface loc (Tags.firstLine src (Range start end)) Nothing)
+      gtags t
+
+instance ToTags Java.InterfaceTypeList where
+  tags t@Java.InterfaceTypeList { extraChildren = interfaces } = do
+    src <- ask @Source
+    for_ interfaces $ \x -> case x of
+      Java.Type (Prj (Java.UnannotatedType (Prj (Java.SimpleType ( Prj Java.TypeIdentifier { ann = loc@Loc {byteRange = range}, text = name }))))) ->
+        Tags.yield (Tag name Implementation loc (Tags.firstLine src range) Nothing)
```

Comment:
I'm wondering about the use of `Tags.firstLine` here, and wondering if we'd be able to avoid `ask @Source` by using the `byteRange` from `Java.TypeIdentifier`?

Reply:
I think we already do here? The last parameter of `firstLine` is `range,` where `range = byteRange` in the binding.

Reply:
We still need the `ask @Source` so we can slice out the associated text.

Reply:
@patrickt you're right, yeah I was mistaken in thinking the slicing on the `Source` was happening elsewhere in the tag output. This looks 👍